test/integration/ThreeChainEndToEnd.t.sol

```diff
@@ -0,0 +1,213 @@
+            syncRequestManager: deployC.syncRequestManager(),
+            usdc: new ERC20(6)
+        });
+
+        // Initialize USDC on chain C
+        sC.usdc.file("name", "USD Coin");
+        sC.usdc.file("symbol", "USDC");
+    }
+}
+
+/// @title  Three Chain End-to-End Use Cases
+/// @notice Test cases for the three-chain setup
+contract ThreeChainEndToEndUseCases is ThreeChainEndToEndDeployment {
+    using CastLib for *;
+    using MessageLib for *;
+
+    /// @notice Configure the third chain (C) with assets
+    /// forge-config: default.isolate = true
+    function testConfigureThirdChainAsset() public {
+        sC.usdc.mint(INVESTOR_A, INVESTOR_A_USDC_AMOUNT);
+        sC.spoke.registerAsset{value: GAS}(h.centrifugeId, address(sC.usdc), 0);
+
+        assertEq(h.hubRegistry.decimals(newAssetId(CENTRIFUGE_ID_C, 1)), 6, "expected decimals");
+    }
+
+    /// @notice Configure a pool with support for all three chains
+    /// forge-config: default.isolate = true
+    function testConfigurePoolWithThreeChains() public {
+        // Configure spokes
+        sB.usdc.mint(INVESTOR_A, INVESTOR_A_USDC_AMOUNT);
+        sB.spoke.registerAsset{value: GAS}(h.centrifugeId, address(sB.usdc), 0);
+        sC.usdc.mint(INVESTOR_A, INVESTOR_A_USDC_AMOUNT);
+        sC.spoke.registerAsset{value: GAS}(h.centrifugeId, address(sC.usdc), 0);
+
+        // Create and configure the pool on hub (chain A)
+        vm.prank(address(h.guardian.safe()));
+        h.guardian.createPool(POOL_A, FM, USD_ID);
+
+        vm.startPrank(FM);
+        h.hub.setPoolMetadata(POOL_A, bytes("Testing pool with three chains"));
+        h.hub.addShareClass(POOL_A, "Tokenized MMF", "MMF", bytes32("salt"));
+
+        h.hub.notifyPool{value: GAS}(POOL_A, sB.centrifugeId);
+        h.hub.notifyPool{value: GAS}(POOL_A, sC.centrifugeId);
+
+        h.hub.notifyShareClass{value: GAS}(POOL_A, SC_1, sB.centrifugeId, sB.redemptionRestrictionsHook.toBytes32());
+        h.hub.notifyShareClass{value: GAS}(POOL_A, SC_1, sC.centrifugeId, sC.redemptionRestrictionsHook.toBytes32());
+
+        h.hub.createAccount(POOL_A, ASSET_ACCOUNT, true);
+        h.hub.createAccount(POOL_A, EQUITY_ACCOUNT, false);
+        h.hub.createAccount(POOL_A, LOSS_ACCOUNT, false);
+        h.hub.createAccount(POOL_A, GAIN_ACCOUNT, false);
+
+        AssetId USDC_ID_B = newAssetId(sB.centrifugeId, 1);
+        AssetId USDC_ID_C = newAssetId(sC.centrifugeId, 1);
+
+        h.hub.initializeHolding(
+            POOL_A, SC_1, USDC_ID_B, h.identityValuation, ASSET_ACCOUNT, EQUITY_ACCOUNT, GAIN_ACCOUNT, LOSS_ACCOUNT
+        );
+        h.hub.initializeHolding(
+            POOL_A, SC_1, USDC_ID_C, h.identityValuation, ASSET_ACCOUNT, EQUITY_ACCOUNT, GAIN_ACCOUNT, LOSS_ACCOUNT
+        );
+
+        h.hub.updateBalanceSheetManager{value: GAS}(sB.centrifugeId, POOL_A, BSM.toBytes32(), true);
+        h.hub.updateBalanceSheetManager{value: GAS}(sC.centrifugeId, POOL_A, BSM.toBytes32(), true);
```

Comment:
Same here (?)

Reply:
Eventually I decided to keep it as part of `configurePool` which is now part of `EndToEndUtils` and generic for a given spoke chain.